### ISSUE TITLE
Idea: Simplify daemon configuration from webui

### ISSUE THREAD

**Issue by Gozala** [ipfs-shipyard/ipfs-desktop#790]
At the moment you have a pretty non user friendly way to configure webui that requires user to:

1. CORS are configured appropriately.
2. Daemon is running.
3. It uses default `5001` port.

And take appropriate actions usually requiring running things in terminal, and possibly entering some results back to webui.

### Idea

I like to propose improved flow. Desktop apps can register themself as handlers of custom protocol, and that could be used to say register `ipfsdaemon:` protocol and have link in the in the webui that once clicked will trigger daemon that can in return encode it's config like `port` into query string and open triggering page with that query, letting webui know how to reconfigure itself.

You can use similar idea to request daemon to change CORS settings as well.

**Comment by hacdias** [comment]
Hey @Gozala! Thanks for the feedback. However, I'm just confused about the following:

> (...) requires user to (...) CORS are configured appropriately.

If you open the Web UI inside IPFS Desktop, you shouldn't need to take any manual action since it doesn't need to set up any CORS headers. Only if you want to open it in a browser.

> (...) requires user to (...) Daemon is running.

IPFS Desktop does that for you.

> (...) requires user to (...) It uses default 5001 port.

Yes, that's the default, but if you change it, IPFS Desktop shall not have any troubles figuring out the port since we send the complete API address with host and port.

---

So I'm just wondering if this issues are related to IPFS Desktop or only about Web UI itself, when decoupled from IPFS Desktop app.

**Comment by Gozala** [comment]
Hi @hacdias I think this issue was probably more appropriate at https://github.com/ipfs/ipfs-gui than here. That being said here additional details:

> If you open the Web UI inside IPFS Desktop, you shouldn't need to take any manual action since it doesn't need to set up any CORS headers. Only if you want to open it in a browser.

If IPFS daemon isn't running or non default configuration is used this is what you'd see when you open webui with companion extension:

<img width="852" alt="screen shot 2019-01-14 at 4 04 57 pm" src="https://user-images.githubusercontent.com/21236/51149315-6ab36680-1816-11e9-8791-384eedd37ab4.png">

And this is what you will see if you go to https://webui.ipfs.io/

<img width="852" alt="screen shot 2019-01-14 at 4 05 10 pm" src="https://user-images.githubusercontent.com/21236/51149331-730ba180-1816-11e9-8f81-e836e45b5bae.png">

Also later is true if daemon is configuration does not include `https://webui.ipfs.io/` in the `Access-Control-Allow-Origin` (or `*` where later is a terrible idea as any https site will be able to control your IPFS daemon).

**Comment by Gozala** [comment]
>
>> (...) requires user to (...) Daemon is running.
>
> IPFS Desktop does that for you.

Yes it does if you have it installed and if it is running. However, if desktop isn't running or is configured differently you'll see the screen shown in previous comment. And user needs instruction to launch `ipfs-desktop` (and if it's configured as default) it should all work out, if it's not then user needs to copy & paste commands from the instructions into terminal to get it working.

Which is why I suggested to make desktop app register itself as default handler of the custom protocol like `ipfsdaemon:` as that would allow replacing instructions with single link / button that will launch an app and pass config to it.

**Comment by Gozala** [comment]
>
>>    (...) requires user to (...) It uses default 5001 port.
>
> Yes, that's the default, but if you change it, IPFS Desktop shall not have any troubles figuring out the port since we send the complete API address with host and port.

That only covers window provided by desktop and it does not cover companion or the https://webui.ipfs.io which is what my proposal was geared towards.

**Comment by hacdias** [comment]
Hey @Gozala! I get your point, but I think you're probably confusing Web UI with our Desktop (menubar) app:

![image](https://user-images.githubusercontent.com/5447088/51169971-21f6be80-18a5-11e9-9271-1a17ebc3b216.png)

When you install IPFS Desktop, it also installs IPFS for you. If you install it and click, for example, on Status, it will open an Web UI window configured for you with no need for further configuration.

**Comment by Gozala** [comment]
@hacdias I’m aware of that. I know that gui can be accessed through electron window that is open when you click it. However, I still would argue that it is important to address other user flows of accessing GUI through companion addon or a URL.

**Comment by hacdias** [comment]
Yes, that's true. I'm just moving it out of this repository since it doesn't actually _fit_ here.
